network/endpoint_test.go

```diff
@@ -5,20 +5,192 @@ package network
+					Expect(err).To(Equal(errEndpointNotFound))
+					Expect(ep).To(BeNil())
+				})
+			})
+
+			Context("When one endpoint found", func() {
+				It("Should return endpoint", func() {
+					podName := "test"
+					podNS := "ns"
+					nw := &network{
+						Endpoints: map[string]*endpoint{},
+					}
+					nw.Endpoints["pod"] = &endpoint{
+						PODName: podName,
+						PODNameSpace: podNS,
+					}
+					ep, err := nw.getEndpointByPOD(podName, podNS, true)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(ep.PODName).To(Equal(podName))
+				})
+			})
+		})
+
+		Describe("Test podNameMatches", func() {
+
+			Context("When doExactMatch", func() {
+				It("Should exact match", func() {
+					actual := "nginx"
+					valid := "nginx"
+					invalid := "nginx-deployment-5c689d88bb"
+					Expect(podNameMatches(valid, actual, true)).To(BeTrue())
+					Expect(podNameMatches(invalid, actual, true)).To(BeFalse())
+				})
+			})
+
+			Context("When not doExactMatch", func() {
+				It("Should not exact match", func() {
+					actual := "nginx"
+					valid1 := "nginx"
+					valid2 := "nginx-deployment-5c689d88bb"
+					invalid := "nginx-deployment-5c689d88bb-qwq47"
+					Expect(podNameMatches(valid1, actual, false)).To(BeTrue())
+					Expect(podNameMatches(valid2, actual, false)).To(BeTrue())
+					Expect(podNameMatches(invalid, actual, false)).To(BeFalse())
+				})
+			})
+		})
+
+		Describe("Test attach", func() {
+
+			Context("When SandboxKey in use", func() {
+				It("Should raise errEndpointInUse", func() {
+					ep := &endpoint{
+						SandboxKey: "key",
+					}
+					err := ep.attach("")
+					Expect(err).To(Equal(errEndpointInUse))
+				})
+			})
+
+			Context("When SandboxKey now in use", func() {
+				It("Should set SandboxKey", func() {
+					sandboxKey := "key"
+					ep := &endpoint{}
+					err := ep.attach(sandboxKey)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(ep.SandboxKey).To(Equal(sandboxKey))
+				})
+			})
+		})
+
+		Describe("Test detach", func() {
+
+			Context("When SandboxKey not in use", func() {
+				It("Should raise errEndpointNotInUse", func() {
+					ep := &endpoint{}
+					err := ep.detach()
+					Expect(err).To(Equal(errEndpointNotInUse))
+				})
+			})
+
+			Context("When SandboxKey in use", func() {
+				It("Should set SandboxKey empty", func() {
+					ep := &endpoint{
+						SandboxKey: "key",
+					}
+					err := ep.detach()
+					Expect(err).NotTo(HaveOccurred())
+					Expect(ep.SandboxKey).To(BeEmpty())
+				})
+			})
+		})
+
+		Describe("Test updateEndpoint", func() {
+			Context("", func() {
+				It("", func() {
+
+				})
+			})
+		})
+
+		Describe("Test GetPodNameWithoutSuffix", func() {
+			Context("When podnames have suffix or not", func() {
+				It("Should return podname without suffix", func() {
+					testData := map[string]string{
+						"nginx-deployment-5c689d88bb":       "nginx",
+						"nginx-deployment-5c689d88bb-qwq47": "nginx-deployment",
+						"nginx": "nginx",
+					}
```

Comment:
```suggestion
					testData := map[string]string{
						"nginx-deployment-5c689d88bb":              "nginx",
						"nginx-deployment-5c689d88bb-qwq47": "nginx-deployment",
						"nginx" :                                                       "nginx",
					}
```

Reply:
I guess you need a lint tool. Could you please run `gofmt` or more  in the first place?

Reply:
This is copied from the origin test cases, I'll format it

Reply:
seems like we need a lint check in CI.